packages/js/src/plugins/auctionHouseModule/operations/deposit.ts

```diff
@@ -0,0 +1,188 @@
+import { ConfirmOptions, PublicKey } from '@solana/web3.js';
+import type { Metaplex } from '@/Metaplex';
+import { TransactionBuilder } from '@/utils';
+import {
+  createAuctioneerDepositInstruction,
+  createDepositInstruction,
+  DepositInstructionAccounts,
+} from '@metaplex-foundation/mpl-auction-house';
+import {
+  useOperation,
+  Operation,
+  OperationHandler,
+  Signer,
+  toPublicKey,
+  lamports,
+  isSigner,
+  amount,
+  SolAmount,
+  SplTokenAmount,
+} from '@/types';
+import { SendAndConfirmTransactionResponse } from '../../rpcModule';
+import { findAssociatedTokenAccountPda } from '../../tokenModule';
+import { AuctionHouse } from '../models';
+import { findAuctioneerPda, findAuctionHouseBuyerEscrowPda } from '../pdas';
+import { AuctioneerAuthorityRequiredError } from '../errors';
+
+// -----------------
+// Operation
+// -----------------
+
+const Key = 'DepositOperation' as const;
+
+/**
+ * @group Operations
+ * @category Constructors
+ */
+export const depositOperation = useOperation<DepositOperation>(Key);
+
+/**
+ * @group Operations
+ * @category Types
+ */
+export type DepositOperation = Operation<
```

Comment:
In fact, we should probably be slightly more explicit on the client methods as well since we'll be withdrawing from other sources too (fee and treasury accounts).

So instead of `metaplex.auctionHouse().deposit(...)`, let's do `metaplex.auctionHouse().depositToBuyerAccount(...)`. And then we'll have `metaplex.auctionHouse().withdrawFromX(...)` where `X` will be `TreasuryAccount`, `FeeAccount` and `BuyerAccount`.